File: app/src/main/java/com/yibao/music/adapter/SearchPagerAdapter.java
package com.yibao.music.adapter;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentActivity;
import androidx.viewpager2.adapter.FragmentStateAdapter;

import com.yibao.music.fragment.SearchFragment;
import com.yibao.music.fragment.SongCategoryFragment;

/**
 * 作者：Stran on 2017/3/23 03:31
 * 描述：${TODO}
 * 邮箱：strangermy@outlook.com
 *
 * @author Stran
 */
public class SearchPagerAdapter
        extends FragmentStateAdapter {

    private String mArtist;
    public SearchPagerAdapter(@NonNull FragmentActivity fragmentActivity,String artist) {
        super(fragmentActivity);
        mArtist = artist;
    }

    @NonNull
    @Override
    public Fragment createFragment(int position) {
        return SearchFragment.newInstance(position,mArtist);
    }

    @Override
    public int getItemCount() {
        return 4;
    }

}

File: app/src/main/java/com/yibao/music/adapter/SearchPagerAdapter.kt
package com.yibao.music.adapter

import androidx.fragment.app.Fragment
import com.yibao.music.fragment.SearchFragment.Companion.newInstance
import androidx.fragment.app.FragmentActivity
import androidx.viewpager2.adapter.FragmentStateAdapter
import com.yibao.music.fragment.SearchFragment
import com.yibao.music.viewmodel.SearchViewModel

/**
 * 作者：Stran on 2017/3/23 03:31
 * 描述：${TODO}
 * 邮箱：strangermy@outlook.com
 *
 * @author Stran
 */
class SearchPagerAdapter(
    fragmentActivity: FragmentActivity,
    private val artist: String,
    private val viewModel: SearchViewModel
) :
    FragmentStateAdapter(fragmentActivity) {
    override fun createFragment(position: Int): Fragment {
        return newInstance(position, artist, viewModel)
    }

    override fun getItemCount(): Int {
        return 4
    }
}